pkg/modprovider/server.go

```diff
@@ -129,61 +130,99 @@ func dirExists(path string) bool {
 	return false
 }
 
+func extractConfigParamFromArgs(args []string) ([]string, string, bool) {
+	for i, arg := range args {
+		if (arg == "--config" || arg == "-c") && i+1 < len(args) {
+			return args[:i], args[i+1], true
+		}
+	}
+	return args, "", false
+}
+
+func unmarshallConfigFile(configFilePath string, packageName string) (*ModuleConfig, error) {
+	file, err := os.ReadFile(configFilePath)
+	if err != nil {
+		return nil, fmt.Errorf("failed to read config file %s: %w", configFilePath, err)
+	}
+
+	modified := bytes.ReplaceAll(file, []byte("[packageName]"), []byte(packageName))
```

Comment:
This is a little interesting. Needed?

Reply:
It is needed because type references have the package name in their type token but the package name is only available after the user have specified it when adding the package so in order to author these schema files without knowing the package name before I decided to introduce the placeholder `[packageName]`